lib/codecs/corejson.js

```diff
@@ -1,12 +1,22 @@
 const document = require('../document')
 
+function primitiveToNode (data) {
+  if (data instanceof Object && data._type === 'document') {
```

Comment:
Instance of is non-standard & slow. And will return true for arrays (not sure if you want that).